cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java

```diff
@@ -136,6 +148,9 @@ public void error(Throwable cause) {
         throw (Exception) cause;
       }
       throw Throwables.propagate(cause);
+    } catch (InterruptedException e) {
+      Futures.getUnchecked(controller.stop());
```

Comment:
If the stop fails, then an unchecked exception will be thrown instead of the `InterruptedException`. Is that what you wanted?